duckduckgo/Android
# Favourites onboarding experiment

Task/Issue URL: https://app.asana.com/0/488551667048375/1200765290066006/f
Tech Design URL: 
CC: 

**Description**:
This PR adds a new step in our app onboarding to encourage users to try adding a Favorite.

**Steps to test this PR**:

> Happy path
1. Fresh install (force variant to zo)
2. Go to the browser
3. Visit a non SERP site
4. Open a new tab or clear data
5. Ensure favourites onboarding appears
6. Ensure first visited site appears as a Favorite
6. Now click on “Add favorite” (pixel: `m_odc_ok`), or on “+” item (pixel: `m_fav_ai`)
7. Ensure CTA dissmissed
8. Ensure URL focused
9. Now visit a site
10. When page finishes, ensure site was added as favorite (pixel: `m_fav_ad`)

 
> Undo site from onboarding
1. Fresh install (force variant to zo)
2. Go to the browser
3. Visit a non SERP site
4. Open a new tab or clear data
5. Ensure favourites onboarding appears
6. Ensure first visited site appears as a Favorite
7. Click on “Undo” (pixel: `m_fo_u`)
8. Ensure CTA disappears and favorite is removed

> Ensure user event data removed
1. Go to `FavoritesOnboardingWorkRequestBuilder` and change line `36` to 20 seconds
1. Fresh install (force variant to zo)
2. Go to the browser
3. Visit a non SERP site
4. Wait for 20 seconds
5. Open a new tab
6. Ensure favourites onboarding does not appear

> Existing onboarding
1. Fresh install (force variant to zp, which is control group)
2. Do similar steps as in previous test to ensure favourites onboarding does not appear


> Test migration:
1. Install app from develop
2. Open the app
3. Checkout this branch
4. Open the app
5. Ensure migration succeeds


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
